src/main/java/duke/Parser.java

```diff
@@ -0,0 +1,202 @@
+                ui.quit();
+            } else if (command.equals("list")) {
+                ui.listOutTasks(taskList);
+            } else {
+                String[] splitStr = command.split(" ");
+                Task taskToMark = getTaskToMark(splitStr, taskList);
+                Task taskToAdd = getTaskToAdd(command);
+                Task taskToDelete = getTaskToDelete(splitStr, taskList);
+                List<Task> taskListWithKeyWord = getAllTaskWithKeyword(command, taskList);
+                if (taskToMark != null) {
+                    String action = splitStr[0];
+                    if (action.equals("mark")) {
+                        ui.markTask(taskToMark);
+                    } else {
+                        ui.unmarkTask(taskToMark);
+                    }
+                } else if (taskToAdd != null) {
+                    ui.addTask(taskList, taskToAdd);
+                } else if (taskToDelete != null) {
+                    ui.deleteTask(taskList, taskToDelete);
+                } else if (taskListWithKeyWord.size() != 0) {
+                    ui.printTasksWithKeyword(taskListWithKeyWord);
+                } else {
+                    throw new DukeException("☹ OOPS!!! I'm sorry, but I don't know what that means :-(\n");
+                }
+            }
+        } catch (DukeException de) {
+            System.out.println(de);
+        }
+    }
+
+    /**
+     * This method takes in the string input of the user
+     * and return the task from the taskList to be deleted.
+     * When an invalid number is entered, user will get to
+     * enter again
+     * @param splitStr This is the user's string input split with " " into an array
+     * @param taskList This is the user's list of tasks
+     * @return Task The task from the user's task list to be deleted
+     * @throws DukeException This tells user to enter a valid number when an invalid task number is entered
+     */
+    private static Task getTaskToDelete(String[] splitStr, TaskList taskList) throws DukeException {
+        try {
+            if (splitStr.length != 2) {
+                return null;
+            }
+            int index = Integer.parseInt(splitStr[1]);
+            String action = splitStr[0];
+            boolean validAction = action.equals("delete");
+            boolean validIndex = index > 0 && index <= taskList.getSize();
+            if (validIndex && validAction) {
+                return taskList.getTask(index - 1);
+            } else if (validAction && !validIndex) {
+                throw new DukeException("☹ OOPS!!! Please enter a valid task number to delete");
+            }
+            return null;
+        } catch (NumberFormatException e) {
+            return null;
+        }
+    }
+
+    /**
+     * This method takes in the string input of the user
+     * and return the task from the taskList to be marked/unmarked.
+     * When an invalid number is entered, user will get to
+     * enter again

```

Comment:
Do consider rephrasing comments according to match java coding standard (i.e. MARKS the task from taskList to be done/undone.)